csm_web/scheduler/models.py

```diff
@@ -36,6 +36,7 @@ def week_bounds(date):
 
 class User(AbstractUser):
     priority_enrollment = models.DateTimeField(null=True, blank=True)
+    subscribed = models.BooleanField(default=True, help_text="Email subscription status")
```

Comment:
I feel like we've had a conversation about the default here, but what was the reasoning for making this default to True?

It's both a good and a bad thing that people will get notifications for everything right off the bat; they'll be informed of all updates, but they'll also be spammed quite a bit (ex. a user may enroll and drop multiple sections and get an email for every single one of these actions, and the mentors will correspondingly be emailed those notifications as well.)

Having this be opt-in could be better for the amount of email traffic we use, and would also prevent unwanted spam from users. We aren't doing any marketing or anything, so we aren't really losing too much by making this be opt-in rather than opt-out.

I presume the concern was that people wouldn't know that this feature exists, but we can still advertise this feature during orientation, in announcements in slack, etc., as this is mostly helpful for mentors, as opposed to students. Maybe when waitlisting and section swapping are implemented, this could be more useful for students, but even then we can have messages displayed to students asking if they want to enable notifications when they waitlist for a section, or when they send out a request for a swap.

Also, there is also an opportunity here to be more fine-grained with the subscription configuration; there could be options for each possible email that is sent out. For example, a mentor that is also a student may not want any notifications about the sections they're enrolling in, but they do want notifications about students enrolling in their section. Or, a student may not want to receive any notifications about enrolling/dropping from sections, but they do want to receive notifications about waitlists or section swaps (when those come out).

Open to thoughts and discussion on this though!

Reply:
Making False!